File: app/src/main/java/com/cliniva/enventory/ui/FragmentContainerActivity.java
package com.cliniva.enventory.ui;

import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

import com.cliniva.enventory.R;
import com.cliniva.enventory.ui.base.BaseActivity;
import com.cliniva.enventory.ui.customer.CustomerFragment;
import com.cliniva.enventory.ui.main.MainActivity;
import com.cliniva.enventory.ui.supplier.SupplierFragment;
import com.cliniva.enventory.ui.transaction.TransactionFragment;
import com.cliniva.enventory.utils.IntentUtils;

public class FragmentContainerActivity extends BaseActivity {

    @Override
    public int getLayoutRes() {
        return R.layout.activity_fragment_container;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int fragmentType = getIntent().getExtras().getInt("FRAGMENT_TYPE_EXTRA", 0);
        onFragmentTransaction(fragmentType);
    }

    private void onFragmentTransaction(int fragmentType){
        FragmentManager manager = getSupportFragmentManager();
        FragmentTransaction transaction = manager.beginTransaction();

        switch (fragmentType){
            case 0:
                transaction.replace(R.id.fl_fragment_container, TransactionFragment.getInstance()).commit();
                break;
            case 1:
                transaction.replace(R.id.fl_fragment_container, CustomerFragment.getInstance()).commit();
                break;
            case 2:
                transaction.replace(R.id.fl_fragment_container, SupplierFragment.getInstance()).commit();
                break;
        }
    }

    @Override
    public void onBackPressed() {
        Bundle bundle = new Bundle();
        bundle.putInt("TRANSACTION_KEY", 0);
        IntentUtils.getInstance().onActivityIntentWithExtras(this, MainActivity.class, bundle);
    }
}

File: app/src/main/java/com/amirsons/inventory/ui/activity/FragmentContainerActivity.kt
package com.amirsons.inventory.ui.activity

import android.os.Bundle
import com.amirsons.inventory.R
import com.amirsons.inventory.ui.base.BaseActivity
import com.amirsons.inventory.ui.fragment.customer.CustomerFragment
import com.amirsons.inventory.ui.fragment.supplier.SupplierFragment
import com.amirsons.inventory.ui.fragment.transaction.TransactionFragment

class FragmentContainerActivity : BaseActivity() {

    override val contentLayout: Int
        get() = R.layout.activity_fragment_container

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        var fragmentType = 1

        if (intent.extras != null) {
            fragmentType = intent.extras!!.getInt("FRAGMENT_TYPE_EXTRA", 1)
        }

        onFragmentTransaction(fragmentType)
    }

    private fun onFragmentTransaction(fragmentType: Int) {

        val manager = supportFragmentManager
        val transaction = manager.beginTransaction()

        when (fragmentType) {

            1 -> transaction.replace(R.id.fl_fragment_container, TransactionFragment.instance).commit()

            2 -> transaction.replace(R.id.fl_fragment_container, CustomerFragment.instance).commit()

            3 -> transaction.replace(R.id.fl_fragment_container, SupplierFragment.instance).commit()
        }
    }

    override fun onBackPressed() {
        finish()
    }
}
